File: m/rls/rlb/imarg.c
/* file -  imarg - get image argument */
#include "m:\rid\rider.h"
/* code -  im_arg - get image argument */
int im_arg(
int arg ,
char *dst ,
int lim ,
int cnt ,
char *(*vec ))
{ char *str ;
  int len = 0;
  *dst = 0;
  if( arg >= cnt)return ( -1 );
  str = vec[arg];
  while ( len < lim) {
    if( (*dst++ = *str++) == 0)break;
    ++len;
  } 
  return ( len);
} 

File: m/rls/imarg.r
file	imarg - get image argument
include	rid:rider

code	im_arg - get image argument

  func	im_arg
	arg : int			; argument number
	dst : * char			; destination buffer
	lim : int			; destination buffer limit
	cnt : int			; argument count
	vec : ** char			; argument vector
	()  : int			; -1 or number of chars in argument
  is	str : * char			;
	len : int = 0			;
	*dst = 0			; terminate buffer
	reply -1 if arg ge cnt		;
	str = vec[arg]			; get the string
	while len lt lim		;
	   *dst++ = *str++		;
	   quit if eq			; all done
	   ++len			; count it
	end				;
	reply len			;
  end
